### PR TITLE
RD2: Prevent Refresh Opportunities when enhanced RDs are enabled

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
Prevent Refresh Opportunities functionality when enhanced Recurring Donations are enabled. 

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata
Apex Class:
- RD_AddDonationsBTN_TEST

Custom Label:
- RD_ErrorAddDonationLimitedToLegacyMode

# Deleted Metadata

### CODE REVIEW

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-039280

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -2016,6 +2016,14 @@ If you&apos;re not sure how to resolve these errors, post a message in the Nonpr
         <shortDescription>Refresh Opportunities</shortDescription>
         <value>Refresh Opportunities</value>
     </labels>
+    <labels>
+        <fullName>RD_ErrorAddDonationLimitedToLegacyMode</fullName>
```
Comment (RadSFDeveloper) [inline src/labels/CustomLabels.labels]: @sam-knox , can you please verify the new custom label?
@meena-sfdc , this is a new label that once reviewed and approved by Sam here, can also be copied into the label freeze PR.

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HrusQAC" target="_blank">W-039280</a>

File: src/classes/CRLP_RecalculateBTN_CTRL.cls [hunk 2]
```diff
@@ -138,18 +138,25 @@ public with sharing class CRLP_RecalculateBTN_CTRL {
...
+
             } else if (this.objType == Account.SObjectType) {
                 RLLP_OppRollup.rollupAccountsFuture(new Set<Id>{ recordId });
 
             } else if (this.objType == General_Accounting_Unit__c.SObjectType) {
                 ALLO_Rollup_SCHED rollup = new ALLO_Rollup_SCHED(recordId);
                 rollup.runRollups();
-            } else if (this.objType == npe03__Recurring_Donation__c.SObjectType) {
+
+            } else if (this.objType == npe03__Recurring_Donation__c.SObjectType
```
Comment (RadSFDeveloper) [inline src/classes/CRLP_RecalculateBTN_CTRL.cls]: When enhanced RDs are enabled and a user disables Customizable Rollups (even though it is not recommended), then legacy rollups are not calculated for the enhanced RDs.
@force2b , @rponti-sforg

Comment (force2b) [reply src/classes/CRLP_RecalculateBTN_CTRL.cls]: Makes sense. The `RD_RecurringDonations.updateRecurringDonationOnOppChange()` method does a similar check and would just return if CRLP is enabled. I wonder if that method should also check for and return if RD2 is enabled.

File: src/labels/CustomLabels.labels (hunk 1 above)
Comment (meena-sfdc) [reply src/labels/CustomLabels.labels]: Thanks, @RadSFDeveloper !

File: src/classes/CRLP_RecalculateBTN_CTRL.cls (hunk 2 above)
Comment (RadSFDeveloper) [reply src/classes/CRLP_RecalculateBTN_CTRL.cls]: The change has been made directly in the method to check for RD2 enablement.

**Comment by RadSFDeveloper** [comment]
@rponti-sforg , the PR is ready for testing.

**Comment by heidmary** [comment]
I checked the Robot failures and although they are not listed as "non-critical" they aren't related to the work in this PR, so merging this PR per discussion with @rponti-sforg earlier today that his testing was complete and this PR can be merged once builds finish.

**Comment by rponti-sforg** [comment]
QE testing is now complete for this PR and associated WI.  Based on WI description, ACs, and PR comments, tests have passed.

For further testing details, see: 
https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001HrusQAC/view
https://salesforce.quip.com/68vtAlQ3IJ9I

PR has been merged.  Post merge builds have passed, except for known, wide spread, Robot test failures.  These will be forwarded to appropriate QE automation resource(s).

**Comment by rponti-sforg** [comment]
> I checked the Robot failures and although they are not listed as "non-critical" they aren't related to the work in this PR, so merging this PR per discussion with @rponti-sforg earlier today that his testing was complete and this PR can be merged once builds finish.

Thank @heidmary for your help.  Appreciate it.

**Pull request merged**